versioned_docs/version-2021sp/lecture5.md

```diff
@@ -3,8 +3,211 @@ id: lecture5
+  <div>
+    <div>
+      You can reach to {name} at
+      <a href={githubLink}>{githubLink}</a>
+    </div>
+  </div>
+);
+```
+
+## Wat! What's going on
+
+### Functional Component
+
+The simplest component in React is a functional component. A functional component
+does not have any internal state. You can think of it as a function whose inputs
+are some JavaScript object and the output is some HTML code that is generated
+from the data.
+
+In React, we call the JavaScript object `props`, so you can see code like this:
+
+```tsx title="ContactCard.tsx"
+import React from 'react';
+
+type Props = { readonly name: string; readonly githubLink: string };
+
+const ContactCard = (props: Props) => (
+  <div>
+    <div>
+      You can reach to {props.name} at
+      <a href={props.githubLink}>{props.githubLink}</a>
+    </div>
+  </div>
+);
+```
+
+Just calling the input `props` is not good for documentation purpose, so we
+usually use the object destruct syntax to make it more explicit:
+
+```tsx title="ContactCard.tsx"
+import React from 'react';
+
+type Props = { readonly name: string; readonly githubLink: string };
+
+const ContactCard = ({ name, githubLink }: Props) => (
+  <div>
+    <div>
+      You can reach to {name} at
+      <a href={githubLink}>{githubLink}</a>
+    </div>
+  </div>
+);
+```
+
+In order for this component to be reused in another file, we need to export it:
+
+```tsx title="ContactCard.tsx"
+import React from 'react';
+
+type Props = { readonly name: string; readonly githubLink: string };
+
+const ContactCard = ({ name, githubLink }: Props) => (
+  <div>
+    <div>
+      You can reach to {name} at
+      <a href={githubLink}>{githubLink}</a>
+    </div>
+  </div>
+);
+
+export default ContactCard;
+```
+
+React does not care about the arrow function syntax, so you can also write
+it in function definition syntax:
+
+```tsx title="ContactCard.tsx"
+import React from 'react';
+
+type Props = { readonly name: string; readonly githubLink: string };
+
+function ContactCard ({ name, githubLink }: Props) = (
+  <div>
+    <div>You can reach to {name} at
+      <a href={githubLink}>{githubLink}</a>
+    </div>
+  </div>
+);
+
+export default ContactCard;
+```
+
+For this course, we prefer that you use arrow function syntax.
+
+### `index.tsx`
+
+```tsx title="index.tsx"
+import React from 'react';
+import ReactDOM from 'react-dom';
+import ContactCard from './ContactCard';
+
+const link = 'https://github.com/cornell-dti';
+const root = document.getElementById('root');
+
+ReactDOM.render(<ContactCard name="Cornell DTI" githubLink={link} />, root);
+```
+
+In a React app, `index.tsx` is usually the entry point. In the code above, it
+renders your component into the DOM element that has id `root`.
+
+## Stateful Component
+
+Imagine you are writing a contacts app and you need to implement an editor.
+
+Unlike the previous components, you need to maintain state. In React, you will
+need `hooks`.
```

Comment:
I think we should explain a bit more indepth what hooks are and also note the fact that `useState` is a hook. Maybe  also mention that the general naming convention for a hook is `useXXXX` so students know what to look out for.

Reply:
that makes sense. Does next lecture go more in-depth about hooks?